### DISCUSSION TITLE
Is it possible to look up a package definition within a *cue.Instance?

### DISCUSSION THREAD

**Discussion by tonyhb** [cue-lang/cue#1571] [Q&A]
I'm doing an `instance.Lookup` and `LookupDef`, both of which fail to look up definitions even with package quantifiers.  It's surprising as the evaluated code uses those packages fine:

```
package repo

import (
    "example.com/foo"
)

fancyValue: foo.#Definition & {}
```

In this case, compiling this into an instance and running `instance.Lookup("foo.#Definition")` fails.  LookupDef also fails with packages as it forces a # symbol prefix, so you can't `instance.LookupDef("foo.#Definition")` - this panics.

The real aim here:  programatically assert that a value subsumes a definition from a package.  ie grab the "fancyValue" field and run `.Subsumes()` within go code.  To do this I need the definition as a value.

**Comment by mpvl** [comment]
`cue.Instance` is deprecated. But you can do so after converting it to a `Value` using the namesake method and then using `LookupPath` with the appropriate values.

**Comment by jlongtine** [comment]
I've got some code that I'm using to valid the shape of a `Value` that's doing something similar. Here's the basic shape of that...

```go
import (
	"cuelang.org/go/cue"
	"cuelang.org/go/cue/cuecontext"
	"cuelang.org/go/cue/load"
	"cuelang.org/go/cue/build"
)


	instances := load.Instances([]string{"example.com/foo"}, &load.Config{})
	inst := instances[0]

	ctx := cuecontext.New()

	value := ctx.BuildInstance(inst)

	schema := value.LookupPath(cue.MakePath(cue.Def("#Definition")))

        // val was previously generated cue.Value... make sure it's from the same Context as instances
	val = val.Unify(planSchema)

```
